### PR TITLE
KnihyDobrovsky scraper

### PR DESCRIPTION
**Pull request by oodoow**

### CODE REVIEW

**Comment by rarous** [comment]
Please, add actors to root `package.json` `workspaces`.

Please, remove ESLint and scripts from actor packages and use repository wide one (call `yarn lint:actors --fix` in repository root to fix any issues).

Please, fill `author` fields in actors `package.json` with your contact e-mail address.

Please, set `license` fields in actors `package.json` to `EPL-2.0`.

I'm working on template and scripts to make this process easier.

**Review by rarous — APPROVED** [review]

**Comment by oodoow** [comment]
> Please, add actors to root `package.json` `workspaces`.
> 
> Please, remove ESLint and scripts from actor packages and use repository wide one (call `yarn lint:actors --fix` in repository root to fix any issues).
> 
> Please, fill `author` fields in actors `package.json` with your contact e-mail address.
> 
> Please, set `license` fields in actors `package.json` to `EPL-2.0`.
> 
> I'm working on template and scripts to make this process easier.
Thanks, it's done.

File: actors/knihydobrovsky-count/main.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/kategorie' });
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/e-knihy', userData: { label: 'SUBLIST' } });
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/audioknihy', userData: { label: 'SUBLIST' } });
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/hry', userData: { label: 'SUBLIST' } });
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/papirnictvi', userData: { label: 'SUBLIST' } });
+    await requestQueue.addRequest({ url: 'https://www.knihydobrovsky.cz/darky', userData: { label: 'SUBLIST' } });
+
+    const proxyConfiguration = await Apify.createProxyConfiguration(
+        {
+            groups: ['RESIDENTIAL'],
```
Comment (metalwarrior665) [inline actors/knihydobrovsky-count/main.js]: Is this necessary? Residentials are fairly expensive.

File: actors/knihydobrovsky-count/src/routes.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const Apify = require('apify');
+
+const { utils: { log } } = Apify;
+
+const completeUrl = (x) => `https://www.knihydobrovsky.cz${x}`;
+
+exports.handleStart = async ({ request, $ }) => {
+    const requestQueue = await Apify.openRequestQueue();
+    const links = $('#main div.row-main li a').not('div:contains("Magnesia Litera")')
+    .map(function () { return $(this).attr('href'); }).get()
```
Comment (metalwarrior665) [inline actors/knihydobrovsky-count/src/routes.js]: You should indent those chained methods so it is clear it is not new statement

File: actors/knihydobrovsky-count/src/routes.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    .filter((x) => !x.includes('magnesia-litera') && !x.includes('velky-knizni-ctvrtek') && !x.includes('knihomanie'));
+    const absoluteLinks = links.map((x) => completeUrl(x));
+    for (const link of absoluteLinks) {
+        await requestQueue.addRequest({ url: link, userData: { label: 'LIST' } });
+    }
+};
+
+exports.handleSubList = async ({ request, $ }) => {
+    const requestQueue = await Apify.openRequestQueue();
```
Comment (metalwarrior665) [inline actors/knihydobrovsky-count/src/routes.js]: It is better to pass the queue via the context than to open it again in each handle

File: actors/knihydobrovsky-count/src/routes.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            categoryCount[getCategoryKey($)] = getNumberOfProducts($);
+        } else {
+            // go next max page
+            const hrefTrimmed = href.trim();
+            await requestQueue.addRequest({ url: completeUrl(hrefTrimmed), userData: { label: 'LIST' } });
+        }
+    }
+};
+
+exports.handleDetail = async ({ request, $ }) => {
```
Comment (metalwarrior665) [inline actors/knihydobrovsky-count/src/routes.js]: Remove empty fn

**Review by metalwarrior665 — COMMENTED** [review]
Just details but please explore the proxy options

**Comment by rarous** [comment]
It's optional, but it would be nice to have proxies and max concurrency configurable via Input. Eg. like this https://github.com/topmonks/hlidac-shopu/blob/trunk/actors/rohlik-detail/main.js#L229-L265

**Review by metalwarrior665 — APPROVED** [review]

**Comment by metalwarrior665** [comment]
@rarous Can you help @oodoow to tag the Dockerfile FROMs correctly? :) I assume just add a node version, right?

**Comment by rarous** [comment]
It's not a blocker, I'll actually disable this check for actors. Base images have only `latest` tag and it is implicit.

**Comment by zpelechova** [comment]
Is it ready to be merged and built please?

**Comment by oodoow** [comment]
> Is it ready to be merged and built please?

Yes please. :)

**Pull request merged**